src/kbmod/analysis/wcs_utils.py

```diff
@@ -0,0 +1,279 @@
+    wcstan.wcs.ctype = ["RA---TAN", "DEC--TAN"]
+    wcstan.wcs.crval = [ref_val.ra.deg, ref_val.dec.deg]
+    return wcstan
+
+
+def solve_wcs_for_fov(wcs, fov):
+    """Solve for the pixel scale of the WCS given the field of view.
+
+    Parameters
+    ----------
+    wcs : astropy.wcs.WCS
+        The WCS to solve for.
+    fov : astropy.units.Quantity
+        The field of view of the first image dimension to solve for.
+
+    Returns
+    -------
+    astropy.wcs.WCS
+        The WCS with the pixel scale adjusted to match the field of view.
+
+    Notes
+    -----
+    This is a slow iterative solution. It should be possible to solve this analytically.
+    The method starts with the pixel scale of the reference pixel and then expands the
+    lower and upper search bounds by a factor of two until the field of view is bracketed.
+    Then a bisection search is performed to find the pixel scale that matches the field of view.
+    An iteration limit is imposed to prevent an infinite loop.
+    """
+    # Assuming that the pixel scale is the arc of the reference pixel.
+    ref_pixel = wcs.wcs.crpix
+    image_size = wcs.array_shape
+    cdelt_original = wcs.wcs.cdelt[1]
+    tolerance = 1e-12
+    iterations = 1000
+
+    refsep2 = _update_calc_fov(wcs, ref_pixel, image_size, cdelt_original)
+
+    # early out check so we don't waste time if the fov is already close.
+    if abs(refsep2[0] - fov).value < tolerance:
+        return wcs
+
+    # start with bounds equal to the pixel scale. If the fov is already close
+    # then this will be the solution.
+    pixel_scale_bounds = [cdelt_original, cdelt_original]
+    # only one of he bounds needs to be expanded
+    if fov <= refsep2[0]:
+        # expand the lower bound by delta (doubling each iteration) until the fov is bracketed
+        delta = cdelt_original / 16.0
+        while 0 < iterations and tolerance < abs(refsep2[0] - fov).value and fov <= refsep2[0]:
+            delta *= 2.0
+            pixel_scale_bounds[0] -= delta
+            refsep2 = _update_calc_fov(wcs, ref_pixel, image_size, pixel_scale_bounds[0])
+            iterations -= 1
+    else:
+        # expand the upper bound by delta (doubling each iteration) until the fov is bracketed
+        delta = cdelt_original / 16.0
+        while 0 < iterations and tolerance < abs(refsep2[0] - fov).value and refsep2[0] < fov:
+            delta *= 2.0
+            pixel_scale_bounds[1] += delta
+            refsep2 = _update_calc_fov(wcs, ref_pixel, image_size, pixel_scale_bounds[1])
+            iterations -= 1
+
+    # assert(pixel_scale_bounds[0] < cdelt_original <= pixel_scale_bounds[1])
+    previous_pixel_scale = cdelt_original
+    pixel_scale = sum(pixel_scale_bounds) / 2.0
+    # assert(previous_pixel_scale != pixel_scale)
+    # The first test will always be true.
+    while 0 < iterations and tolerance < abs(refsep2[0] - fov).value and previous_pixel_scale != pixel_scale:
```

Comment:
Can you add a comment as to what this loop is doing (similar to the comments you have for the two loops above)?

Reply:
added